airflow/models.py

```diff
@@ -343,10 +343,15 @@ def kill_zombies(self, session=None):
                 dag = self.dags[ti.dag_id]
                 if ti.task_id in dag.task_ids:
                     task = dag.get_task(ti.task_id)
+
+                    # now set non db backed vars on ti
                     ti.task = task
-                    ti.handle_failure("{} killed as zombie".format(ti))
+                    ti.test_mode = False
```

Comment:
Sure but that does answer my question. What if test_mode needs to be true? Can that be the case?